File: CodeChef/LongChallenge/AprilLongChallenge2020B/COVIDLQ.js
//https://www.codechef.com/APRIL20B/problems/COVIDLQ
process.stdin.resume();
process.stdin.setEncoding('utf-8');
let arr = "";
process.stdin.on('data', function (chunk) {
    arr += chunk;
});
process.stdin.on('end', function () {

    arr = arr.split("\n");

    let testCases = arr[0];

    for (let i = 1; i <= 2 * testCases; i += 2) {
        let n = arr[i];
        let queue = arr[i + 1].split(" ");
        let isSafe = true;
        let prevOneIndex = -1;

        for (let j = 0; j < queue.length; j++) {

            if (queue[j] === '1') {

                if (prevOneIndex != -1) {

                    if (j - prevOneIndex < 6) {

                        isSafe = !isSafe;
                        break;
                    }
                }
                prevOneIndex = j;
            }
        }

        if (isSafe)
            console.log("YES");
        else
            console.log("NO");
    }




});

File: CodeChef/LongChallenge/AprilLongChallenge2020B/COVIDLQ.cpp
//https://www.codechef.com/APRIL20B/problems/COVIDLQ

#include <bits/stdc++.h>
#define ll long long int
using namespace std;

int main()
{

#ifdef DEBUG
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
#endif

    ios_base ::sync_with_stdio(false);
    cin.tie(0);

    int t, n;

    cin >> t;

    while (t--)
    {
        int prevOneIndex = -1;
        bool isSocialDistancingFollowed = true;
        cin >> n;

        vector<int> v(n);
        for (int i = 0; i < n; i++)
            cin >> v[i];

        for (int i = 0; i < n; i++)
        {
            if (v[i] == 1)
            {

                if (prevOneIndex != -1)
                {
                    if (i - prevOneIndex < 6)
                    {
                        isSocialDistancingFollowed = false;
                        break;
                    }
                }
                prevOneIndex = i;
            }
        }

        if (isSocialDistancingFollowed)
            cout << "YES\n";
        else
            cout << "NO\n";
    }

    return 0;
}